aws/logs_monitoring/lambda_function.py

```diff
@@ -635,6 +635,10 @@ def forward_metrics(metrics):
             if logger.isEnabledFor(logging.DEBUG):
                 logger.debug(f"Forwarded metric: {json.dumps(metric)}")
 
+    # Force the background thread to send metrics
+    # If this does not happen before the function ends, the metrics may not be sent
+    lambda_stats.flush()
```

Comment:
We do flush the metrics (and it blocks) after executing the handler function https://github.com/DataDog/datadog-lambda-python/blob/a1139b4940229821d29a6fd4c20cf06d03e81de4/datadog_lambda/wrapper.py#L141. Perhaps you can add some debugging statements around that line or in the wrapper code to see what's going on?

Reply:
The integration tests were being run against the unwrapped handler 🤬 

I'll fix this in the morning!